core/templates/components/main-nav/main-nav.twig

```diff
@@ -0,0 +1,44 @@
+{%- if markup is not empty %}
```

Comment:
I would move this to just before the for loop (currently line 10). Right now, if I pass in anything to markup, I undo ALL of the main nav. 

As a front end developer I want to be able to pass in my own `<ul>` of navigation items and have the menu "just work". I Don't want to have to think about the wrappers or the collapse button.